Generator emits invalid `=> { ... }` expression body for SystemTextJson format

### Description

The source generator `ZeroAlloc.Serialisation.Generator` produces invalid C# when a type is annotated with `[ZeroAllocSerializable(SerializationFormat.SystemTextJson)]`. The emitted `Serialize` method uses expression-body syntax (`=> ...`) but the body for the JSON case is a statement block (`{ using var _jw = ...; JsonSerializer.Serialize(_jw, value); }`), which is not valid after `=>`.

MemoryPack and MessagePack branches are not affected because their `serializerCall` is a single statement.

### Steps to Reproduce

```csharp
using ZeroAlloc.Serialisation;

namespace Demo;

[ZeroAllocSerializable(SerializationFormat.SystemTextJson)]
public sealed class WeatherResponse
{
    public string City { get; set; } = "";
    public double TemperatureC { get; set; }
}
```

Build the project. The generator emits `WeatherResponseSerializer.g.cs`:

```csharp
internal sealed class WeatherResponseSerializer : ISerializer<Demo.WeatherResponse>
{
    public void Serialize(IBufferWriter<byte> writer, Demo.WeatherResponse value)
        => { using var _jw = new global::System.Text.Json.Utf8JsonWriter(writer); global::System.Text.Json.JsonSerializer.Serialize(_jw, value); }

    public Demo.WeatherResponse? Deserialize(ReadOnlySpan<byte> buffer)
    {
        if (buffer.IsEmpty) return default;
        return global::System.Text.Json.JsonSerializer.Deserialize<Demo.WeatherResponse>(buffer);
    }
}
```

Compile fails with:

```
error CS0825: The contextual keyword 'var' may only appear within a local variable declaration or in script code
```

### Expected Behavior

The generator should emit a regular method body (with braces) for the `SystemTextJson` case instead of an expression body, producing valid C#:

```csharp
public void Serialize(IBufferWriter<byte> writer, Demo.WeatherResponse value)
{
    using var _jw = new global::System.Text.Json.Utf8JsonWriter(writer);
    global::System.Text.Json.JsonSerializer.Serialize(_jw, value);
}
```

### Actual Behavior

Expression-body syntax is applied unconditionally to the `serializerCall` placeholder, so the JSON branch's block expression results in invalid syntax — `=> { ... }`.

### Root Cause

`src/ZeroAlloc.Serialisation.Generator/SerializerEmitter.cs` emits this template for every format:

```csharp
public void Serialize(IBufferWriter<byte> writer, {{model.FullTypeName}} value)
    => {{serializerCall}}
```

…while `serializerCall` for SystemTextJson is:

```csharp
"{ using var _jw = new global::System.Text.Json.Utf8JsonWriter(writer); global::System.Text.Json.JsonSerializer.Serialize(_jw, value); }"
```

The fix is either to (a) switch the SystemTextJson branch to produce a single expression (e.g., wrap in a helper) or (b) emit a regular method body for all branches by always wrapping `serializerCall` with braces and dropping the `=>`.

### ZeroAlloc.Serialisation Version

1.3.1

### .NET Version

10.0.106 (also reproduces on .NET 8)

### Additional Context

Generated file path (example):
\`obj/Debug/net8.0/generated/ZeroAlloc.Serialisation.Generator/ZeroAlloc.Serialisation.Generator.SerializerGenerator/WeatherResponseSerializer.g.cs\`

Also worth noting: the generator filters on `ClassDeclarationSyntax or StructDeclarationSyntax`, so `record class WeatherResponse(...)` is silently skipped. Not a bug per se, but it contributed to the initial diagnosis confusion — worth documenting in the README, or widening the filter to include `RecordDeclarationSyntax`.

--- a/src/ZeroAlloc.Serialisation.Generator/SerializerEmitter.cs
+++ b/src/ZeroAlloc.Serialisation.Generator/SerializerEmitter.cs
@@ -14,7 +14,7 @@ internal static class SerializerEmitter
             "MessagePack" =>
                 "global::MessagePack.MessagePackSerializer.Serialize(writer, value);",
             "SystemTextJson" =>
-                "{ using var _jw = new global::System.Text.Json.Utf8JsonWriter(writer); global::System.Text.Json.JsonSerializer.Serialize(_jw, value); }",
+                "using var _jw = new global::System.Text.Json.Utf8JsonWriter(writer);\n        global::System.Text.Json.JsonSerializer.Serialize(_jw, value);",
             _ => throw new System.InvalidOperationException($"Unknown format: {model.FormatName}"),
         };
 
@@ -46,7 +46,9 @@ internal static class SerializerEmitter
             {{ns}}internal sealed class {{model.TypeName}}Serializer : ISerializer<{{model.FullTypeName}}>
             {
                 public void Serialize(IBufferWriter<byte> writer, {{model.FullTypeName}} value)
-                    => {{serializerCall}}
+                {
+                    {{serializerCall}}
+                }
 
                 public {{model.FullTypeName}}? Deserialize(ReadOnlySpan<byte> buffer)
                 {

--- a/src/ZeroAlloc.Serialisation.Generator/SerializerGenerator.cs
+++ b/src/ZeroAlloc.Serialisation.Generator/SerializerGenerator.cs
@@ -15,7 +15,7 @@ public sealed class SerializerGenerator : IIncrementalGenerator
             .ForAttributeWithMetadataName(
                 AttributeFullName,
                 predicate: static (node, _) =>
-                    node is ClassDeclarationSyntax or StructDeclarationSyntax,
+                    node is TypeDeclarationSyntax,
                 transform: static (ctx, ct) => ModelExtractor.Extract(ctx, ct))
             .Where(static m => m is not null)
             .Select(static (m, _) => m!);

--- a/tests/ZeroAlloc.Serialisation.Generator.Tests/RecordAndJsonBugTests.cs
+++ b/tests/ZeroAlloc.Serialisation.Generator.Tests/RecordAndJsonBugTests.cs
@@ -1,0 +1,139 @@
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Xunit;
+
+namespace ZeroAlloc.Serialisation.Generator.Tests;
+
+public sealed class RecordAndJsonBugTests
+{
+    [Fact]
+    public void SystemTextJson_GeneratesValidBody_WithNoSyntaxErrors()
+    {
+        var source = """
+            using ZeroAlloc.Serialisation;
+            namespace Demo;
+            [ZeroAllocSerializable(SerializationFormat.SystemTextJson)]
+            public sealed class WeatherResponse
+            {
+                public string City { get; set; } = "";
+                public double TemperatureC { get; set; }
+            }
+            """;
+
+        var (generated, diagnostics) = Generate(source);
+        Assert.DoesNotContain(diagnostics, static d => d.Severity == DiagnosticSeverity.Error);
+        Assert.Contains("public void Serialize", generated, System.StringComparison.Ordinal);
+        // Must NOT be an expression body with a block expression (invalid C#)
+        Assert.DoesNotContain("=> {", generated, System.StringComparison.Ordinal);
+        // Compiling the generated source with the original must succeed
+        AssertGeneratedCompiles(source);
+    }
+
+    [Fact]
+    public void RecordClass_GeneratesSerializer()
+    {
+        var source = """
+            using ZeroAlloc.Serialisation;
+            namespace Demo;
+            [ZeroAllocSerializable(SerializationFormat.SystemTextJson)]
+            public sealed record class WeatherResponseRecord(string City, double TemperatureC);
+            """;
+
+        var (generated, _) = Generate(source);
+        Assert.Contains("WeatherResponseRecordSerializer", generated, System.StringComparison.Ordinal);
+        Assert.Contains("ISerializer<Demo.WeatherResponseRecord>", generated, System.StringComparison.Ordinal);
+    }
+
+    [Fact]
+    public void RecordStruct_GeneratesSerializer()
+    {
+        var source = """
+            using ZeroAlloc.Serialisation;
+            namespace Demo;
+            [ZeroAllocSerializable(SerializationFormat.SystemTextJson)]
+            public readonly record struct PointRecord(int X, int Y);
+            """;
+
+        var (generated, _) = Generate(source);
+        Assert.Contains("PointRecordSerializer", generated, System.StringComparison.Ordinal);
+    }
+
+    private static (string generated, IReadOnlyList<Diagnostic> diagnostics) Generate(string source)
+    {
+        var compilation = CreateCompilation(source);
+        var driver = CSharpGeneratorDriver.Create(new SerializerGenerator())
+            .RunGenerators(compilation);
+
+        var result = driver.GetRunResult();
+        var combined = string.Join(
+            "\n\n",
+            result.GeneratedTrees.Select(static t => t.GetText().ToString()));
+        return (combined, result.Diagnostics.ToArray());
+    }
+
+    private static void AssertGeneratedCompiles(string originalSource)
+    {
+        var compilation = CreateCompilation(originalSource);
+        var driver = CSharpGeneratorDriver.Create(new SerializerGenerator())
+            .RunGenerators(compilation);
+
+        var result = driver.GetRunResult();
+
+        // Compile only the per-type serializer output (not DI/Dispatcher which require
+        // Microsoft.Extensions.DependencyInjection at runtime). This is the file that
+        // historically contained the invalid `=> { ... }` expression body.
+        var serializerTrees = result.GeneratedTrees
+            .Where(static t => t.FilePath.EndsWith("Serializer.g.cs", System.StringComparison.Ordinal)
+                && !t.FilePath.Contains("SerializerDispatcher", System.StringComparison.Ordinal)
+                && !t.FilePath.Contains("Extensions", System.StringComparison.Ordinal))
+            .ToList();
+
+        Assert.NotEmpty(serializerTrees);
+
+        // Synthesize global usings that the generator assumes are present (the hosting
+        // project enables ImplicitUsings; the generator output relies on `System` being
+        // in scope for `ReadOnlySpan<byte>`).
+        const string GlobalUsings = "global using System;\n";
+
+        var allTrees = new List<SyntaxTree>(serializerTrees)
+        {
+            CSharpSyntaxTree.ParseText(originalSource),
+            CSharpSyntaxTree.ParseText(GlobalUsings),
+        };
+
+        var references = BuildReferences();
+        var fullCompilation = CSharpCompilation.Create(
+            "FullCompileTest",
+            allTrees,
+            references,
+            new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary, nullableContextOptions: NullableContextOptions.Enable));
+
+        var allDiagnostics = fullCompilation.GetDiagnostics();
+        Assert.DoesNotContain(allDiagnostics, static d => d.Severity == DiagnosticSeverity.Error);
+    }
+
+    private static CSharpCompilation CreateCompilation(string source)
+    {
+        var syntaxTree = CSharpSyntaxTree.ParseText(source);
+        var references = BuildReferences();
+
+        return CSharpCompilation.Create(
+            "TestAssembly",
+            new[] { syntaxTree },
+            references,
+            new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+    }
+
+    private static List<MetadataReference> BuildReferences()
+    {
+        var zeroAllocRef = MetadataReference.CreateFromFile(
+            typeof(ZeroAlloc.Serialisation.ZeroAllocSerializableAttribute).Assembly.Location);
+
+        return new List<MetadataReference>(Basic.Reference.Assemblies.Net100.References.All)
+        {
+            zeroAllocRef,
+        };
+    }
+}

